pyigm/cgm/utils.py

```diff
@@ -274,4 +274,64 @@ def cgm_from_galaxy_igmsystems(galaxy, igmsystems, rho_max=300*u.kpc, dv_max=400
     # Return
     return cgm_list
 
+def covering_fraction(iontable,colthresh,rhobins=[0,150],returncounts=False,**kwargs):
+    """Given some detection threshold, calculate the ion covering fraction
+    within impact parameter bins
 
+    Parameters
+    ----------
+    iontable : Table
+        Output of CGMAbsSurvey.ion_tbl()
+    colthresh : float or None
```

Comment:
Can we alternatively have an EW threshold?

Reply:
This currently operates on the ion tables, which don't have EW.  Indeed, the table contains summed column densities system by system, and I'm not sure EW really is meaningful in the 'absorption system' context..

Reply:
ah, right.  There is a method to generate 
a transition Table and one might wish to 
feed that in sometime, but it can wait 
another day.